dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-fee-record-payment-entity-groups-to-fee-record-payment-groups.test.ts

```diff
@@ -177,41 +274,135 @@ describe('get-utilisation-report-reconciliation-details-by-id.controller helpers
+            payments: [PaymentEntityMockBuilder.forCurrency('GBP').build()],
+          };
+
+          // Act
+          const result = await mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups([group]);
+
+          // Assert
+          expect(result).toHaveLength(1);
+          expect(result[0].dateReconciled).toBeNull();
+          expect(result[0].reconciledByUser).toBeNull();
+        },
+      );
+
+      it('sets the dateReconciled and reconciledByUser to null when there are RECONCILED fee records in the group but one of the fee records is at the READY_TO_KEY status', async () => {
+        // Arrange
+        const feeRecords = [
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'abc123', dateReconciled: new Date('2023') }).build(),
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'abc123', dateReconciled: new Date('2024') }).build(),
+          FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withStatus('READY_TO_KEY').build(),
+        ];
+        const group: FeeRecordPaymentEntityGroup = {
+          feeRecords,
+          payments: [PaymentEntityMockBuilder.forCurrency('GBP').build()],
+        };
+
+        // Act
+        const result = await mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups([group]);
+
+        // Assert
+        expect(result).toHaveLength(1);
+        expect(result[0].dateReconciled).toBeNull();
+        expect(result[0].reconciledByUser).toBeNull();
+      });
+
+      it('sets the dateReconciled and reconciledByUser to the most recently reconciled fee record when all the fee records in the group are at the RECONCILED status', async () => {
+        // Arrange
+        const feeRecords = [
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'abc123', dateReconciled: new Date('2023') }).build(),
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'def456', dateReconciled: new Date('2024') }).build(),
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'ghi789', dateReconciled: new Date('2022') }).build(),
+        ];
+        const group: FeeRecordPaymentEntityGroup = {
+          feeRecords,
+          payments: [PaymentEntityMockBuilder.forCurrency('GBP').build()],
+        };
+
+        when(findTfmUserSpy)
+          .calledWith('def456')
+          .mockResolvedValue({ ...aTfmUser(), firstName: 'John', lastName: 'Smith' });
+
+        // Act
+        const result = await mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups([group]);
+
+        // Assert
+        expect(result).toHaveLength(1);
+        expect(result[0].reconciledByUser).toEqual<FeeRecordReconciledByUser>({ firstName: 'John', lastName: 'Smith' });
+        expect(result[0].dateReconciled).toEqual(new Date('2024'));
+      });
+
+      it('sets the dateReconciled to the most recently reconciled fee record and the reconciledByUser to null when all the fee records in the group are at the RECONCILED status but the latest fee record has no defined user', async () => {
```

Comment:
I'm not fully sure what we an auto-reconciled fee record actually is 😅, but my intended behaviour is hopefully that an auto-reconciled fee record populates the `dateReconciled` field and leaves the `reconciledByUserId` as `null` (discussed with @ChristianMcc) and that we can update that in future as the design requires

Reply:
It is a fee record with zero fees paid to ukef for the period and then 0 or no fixed fee adjustment and also 0 or no premium balance adjustment (most likely to happen if there are multiple fee records for a facility id and the one with no fees gets keying data generated first).

In terms of the code, in the generate keying data event handler for the fee record state machine, we 'update with status' and sometimes that status is reconciled, which is the auto reconciled case. So I was expecting to see a change in that file so that the date reconciled actually gets set for such fee records (but aware I might be missing something!)